sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/thetasketchesAggregates.scala

```diff
@@ -0,0 +1,652 @@
+        UnionAggregationBuffer(existingUnion)
+      case (UnionAggregationBuffer(union1), UnionAggregationBuffer(union2)) =>
+        union1.union(union2.getResult)
+        UnionAggregationBuffer(union1)
+      // CREATE new union only when necessary
+      case (UpdatableSketchBuffer(sketch1), UpdatableSketchBuffer(sketch2)) =>
+        createUnionWith(sketch1.compact, sketch2.compact)
+      case (UpdatableSketchBuffer(sketch1), FinalizedSketch(sketch2)) =>
+        createUnionWith(sketch1.compact, sketch2)
+      // Should never make it here, but added cases for defensive programming
+      case (FinalizedSketch(sketch1), UpdatableSketchBuffer(sketch2)) =>
+        createUnionWith(sketch1, sketch2.compact)
+      case (FinalizedSketch(sketch1), FinalizedSketch(sketch2)) =>
+        createUnionWith(sketch1, sketch2)
+      case _ => throw QueryExecutionErrors.thetaInvalidInputSketchBuffer(prettyName)
+    }
+  }
+
+  /**
+   * Returns a Compact sketch derived from the input column or expression
+   *
+   * @param sketchState
+   *   Union instance used as an aggregation buffer
+   * @return
+   *   A Compact binary sketch
+   */
+  override def eval(sketchState: ThetaSketchState): Any = {
+    sketchState.eval()
+  }
+
+  /** Convert the underlying UpdateSketch/Union into an Compact byte array */
+  override def serialize(sketchState: ThetaSketchState): Array[Byte] = {
+    sketchState.serialize()
+  }
+
+  /** Wrap the byte array into a Compact sketch instance */
+  override def deserialize(buffer: Array[Byte]): ThetaSketchState = {
+    if (buffer.nonEmpty) {
+      FinalizedSketch(CompactSketch.heapify(Memory.wrap(buffer)))
+    } else {
+      this.createAggregationBuffer()
+    }
+  }
+}
+
+/**
+ * The ThetaUnionAgg function ingests and merges Datasketches ThetaSketch instances previously
+ * produced by the ThetaSketchAgg function, and outputs the merged ThetaSketch.
+ *
+ * See [[https://datasketches.apache.org/docs/Theta/ThetaSketches.html]] for more information.
+ *
+ * @param left
+ *   Child expression against which unique counting will occur
+ * @param right
+ *   the log-base-2 of nomEntries decides the number of buckets for the sketch
+ */
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = """
+    _FUNC_(expr, lgNomEntries) - Returns the ThetaSketch's Compact binary representation.
+      `lgNomEntries` (optional) the log-base-2 of Nominal Entries, with Nominal Entries deciding
+      the number buckets or slots for the ThetaSketch.""",
+  examples = """
+    Examples:
+      > SELECT theta_sketch_estimate(_FUNC_(sketch)) FROM (SELECT theta_sketch_agg(col) as sketch FROM VALUES (1) tab(col) UNION ALL SELECT theta_sketch_agg(col, 20) as sketch FROM VALUES (1) tab(col));
+       1
+  """,
+  group = "agg_funcs",
+  since = "4.1.0")
+// scalastyle:on line.size.limit
+case class ThetaUnionAgg(
+    left: Expression,
+    right: Expression,
+    mutableAggBufferOffset: Int = 0,
```

Comment:
Let's remove the default values for these parameters to protect ourselves against someone forgetting to assign them properly in the future.

Reply:
I understand. Just want to point out that I see they are assigned default values in a couple other aggregation classes in this folder which is why I followed the pattern.